### ISSUE TITLE
Minor cleanup of pam tests in `twisted.test.test_newcred`.

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6316]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)| @tomprince reported|
|-|-|
|Trac ID|trac#6316|
|Type|enhancement|
|Created|2013-02-21 07:37:57Z|
|Branch|https://github.com/twisted/twisted/tree/pam-test-cleanup-6316|

There is a pyflakes error, as well as an open-coded `TestCase.patch`. This cleans them up.

<details><summary>Searchable metadata</summary>

```
trac-id__6316 6316
type__enhancement enhancement
reporter__tom_prince tom.prince
priority__normal normal
milestone__ 
branch__branches_pam_test_cleanup_6316 branches/pam-test-cleanup-6316
branch_author__tomprince tomprince
status__closed closed
resolution__fixed fixed
component__core core
keywords__easy easy
time__1361432277000000 1361432277000000
changetime__1364318005000000 1364318005000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_1"></a>|@tomprince commented|
|-|-|

(In [37165]) Branching to pam-test-cleanup-6316.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_2"></a>|@tomprince commented|
|-|-|

The official site for PyPAM seems to be down. I grabbed a copy for testing from the [gentoo mirrors](http://distfiles.gentoo.org/distfiles/PyPAM-0.5.0.tar.gz).

Note, these tests don't really do anything useful (See [ticket:3278#comment:4 [#3278](https://github.com/twisted/twisted/issues/3278)])

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

[#3278](https://github.com/twisted/twisted/issues/3278) is a ticket about AMP.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_4"></a>|@tomprince commented|
|-|-|

I guess I meant [#3728](https://github.com/twisted/twisted/issues/3728).

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_5"></a>|@wallrj set owner to @wallrj<br>@wallrj set status to `assigned`|
|-|-|

Reviewing...

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_6"></a>|@wallrj removed owner<br>@wallrj set status to `new`|
|-|-|

Code Review:

 * [Build Results look ok - some warnings because branch is out of date](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/pam-test-cleanup-6316)
 * Merges cleanly to trunk
 * The pyflakes unused import warning has gone.
   ```
    [richard@zorin trunk]$ pyflakes twisted/test/test_newcred.py
    twisted/test/test_newcred.py:21: redefinition of unused 'crypt' from line 19
    twisted/test/test_newcred.py:24: 'callIntoPAM' imported but unused

    [richard@zorin pam-test-cleanup-6316]$ pyflakes twisted/test/test_newcred.py
    twisted/test/test_newcred.py:21: redefinition of unused 'crypt' from line 19
    twisted/test/test_newcred.py:26: redefinition of unused 'pamauth' from line 24
   ```
 * I can't actually test this.
   * pamauth expects a "PAM" package. But python-pam provides a "pam" package
     ```
     >>> from twisted.cred import pamauth
     Traceback (most recent call last):
       File "<stdin>", line 1, in <module>
       File "twisted/cred/pamauth.py", line 9, in <module>
         import PAM
     ImportError: No module named PAM
     ```
   * This launchpad bug
     https://bugs.launchpad.net/keystone/+bug/938801 suggests that
     certain distros provide a PAM package...but I can't install that
     either ...pycon wireless issues.

One comment:
 1. You're still saving the original method? Doesn't patch do
    all that for you?
    ```
        self._oldCallIntoPAM = pamauth.callIntoPAM
        pamauth.callIntoPAM = self.callIntoPAM
    ```

Merge after removing those unnecessary lines.

-RichardW.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_7"></a>|@tomprince commented|
|-|-|

I believe [this](http://web.archive.org/web/20110316070059/http://www.pangalactic.org/PyPAM/) is the binding.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_8"></a>|@tomprince set status to `closed`|
|-|-|

(In [37817]) Merge pam-test-cleanup-6316: Cleanup pam tests.

Author: tom.prince
Reviewers: rwall
Fixes: [#6316](https://github.com/twisted/twisted/issues/6316)

There is a pyflakes error, as well as an open-coded
TestCase.patch. This cleans them up.
